src/python/pants/backend/url_handlers/s3/register.py

```diff
@@ -72,9 +72,14 @@ async def download_from_s3(request: S3DownloadFile, aws_credentials: AWSCredenti
     if request.query:
         path_style_url += f"?{request.query}"
 
+    headers = {}
+    if aws_credentials.creds.token:
+        # Workaround https://github.com/boto/botocore/pull/2948
+        headers['x-amz-security-token'] = None
```

Comment:
this doesn't seem to work with your suggested upstream fix, unless botocore wraps the headers in a case insensitive mapping type behind the scenes?